core/Plugin/Manager.php

```diff
@@ -316,18 +318,71 @@ public function isPluginLoaded($name)
      */
     public function readPluginsDirectory()
     {
-        $pluginsName = _glob(self::getPluginsDirectory() . '*', GLOB_ONLYDIR);
         $result = array();
-        if ($pluginsName != false) {
-            foreach ($pluginsName as $path) {
-                if (self::pluginStructureLooksValid($path)) {
-                    $result[] = basename($path);
+        foreach (self::getPluginsDirectories() as $pluginsDir) {
+            $pluginsName = _glob($pluginsDir . '*', GLOB_ONLYDIR);
+            if ($pluginsName != false) {
+                foreach ($pluginsName as $path) {
+                    if (self::pluginStructureLooksValid($path)) {
+                        $result[] = basename($path);
+                    }
                 }
             }
         }
+
+        sort($result);
+
         return $result;
     }
 
+    /**
+     * Returns the path to all plugins directories. Each plugins directory may contain several plugins.
+     * All paths have a trailing slash '/'.
+     * @return array
+     */
+    public static function getPluginsDirectories()
+    {
+        $dirs = array(PIWIK_INCLUDE_PATH . '/plugins/');
+        if (!empty($GLOBALS['MATOMO_PLUGIN_DIRS'])) {
+            $GLOBALS['MATOMO_PLUGIN_DIRS'] = array_map(function ($dir) {
+                return rtrim($dir, '/') . '/';
+            }, $GLOBALS['MATOMO_PLUGIN_DIRS']);
+            $dirs = array_merge($dirs, $GLOBALS['MATOMO_PLUGIN_DIRS']);
```

Comment:
Is it possible to initialize `$GLOBALS['MATOMO_PLUGIN_DIRS']` from an environment variable? Could be useful for some use cases.